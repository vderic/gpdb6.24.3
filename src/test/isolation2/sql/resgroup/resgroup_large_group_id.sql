-- Test resgroup oid larger than int32.
create extension if not exists gp_inject_fault;

select gp_inject_fault('bump_oid', 'skip', dbid) from gp_segment_configuration where role = 'p' and content = -1;

create resource group rg_large_oid with (cpu_rate_limit=20, memory_limit=10);

select gp_inject_fault('bump_oid', 'reset', dbid) from gp_segment_configuration where role = 'p' and content = -1;

select max(oid)::bigint > (power(2,31) + 1)::bigint from pg_resgroup;

-- count(*) > 0 to run the SQL but do not display the result
select count(*) > 0 from pg_resgroup_get_status(NULL);

drop resource group rg_large_oid;
