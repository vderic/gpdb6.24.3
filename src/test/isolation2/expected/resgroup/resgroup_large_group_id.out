-- Test resgroup oid larger than int32.
create extension if not exists gp_inject_fault;
CREATE

select gp_inject_fault('bump_oid', 'skip', dbid) from gp_segment_configuration where role = 'p' and content = -1;
 gp_inject_fault 
-----------------
 Success:        
(1 row)

create resource group rg_large_oid with (cpu_rate_limit=20, memory_limit=10);
CREATE

select gp_inject_fault('bump_oid', 'reset', dbid) from gp_segment_configuration where role = 'p' and content = -1;
 gp_inject_fault 
-----------------
 Success:        
(1 row)

select max(oid)::bigint > (power(2,31) + 1)::bigint from pg_resgroup;
 ?column? 
----------
 t        
(1 row)

-- count(*) > 0 to run the SQL but do not display the result
select count(*) > 0 from pg_resgroup_get_status(NULL);
 ?column? 
----------
 t        
(1 row)

drop resource group rg_large_oid;
DROP
